app/screens/channel_info/channel_info.js

```diff
@@ -176,6 +178,33 @@ export default class ChannelInfo extends PureComponent {
         this.handleDeleteOrLeave('delete');
     };
 
+    handleConvertToPrivate = preventDoubleTap(() => {
+        const {actions, currentChannel} = this.props;
+        const {formatMessage} = this.context.intl;
+        const displayName = {displayName: currentChannel.display_name};
+        const title = {id: t('mobile.channel_info.alertTitleConvertChannel'), defaultMessage: 'Convert {displayName} to Private?'};
+        const message = {
+            id: t('mobile.channel_info.alertMessageConvertChannel'),
+            defaultMessage: 'When you convert {displayName} to a private channel, history and membership are preserved.\n\nPublicly shared files remain accessible to anyone with the link. Membership in a private channel is by invitation only.\n\nThe change is permanent and cannot be undone.',
+        };
+        const onPressAction = () => {
+            actions.convertChannelToPrivate(currentChannel.id).then(() => {
+                this.close();
```

Comment:
What happens if this action fails? and why is this closing? I think we should give feedback in both cases, when it succeeded and if it fails. And in both cases it should stay in the current screen.

Reply:
Thank you for the feedback :)

I did a bit of shameless copying of what [this](https://github.com/mattermost/mattermost-mobile/blob/master/app/screens/channel_info/channel_info.js#L195) method handles its alerts, but it did feel like some error handling should be in place.

Also, what would be a good way to provide the user feedback on the request succeeding or failing? Are there any good references in the mobile app I could refer to when setting up this behavior?

Reply:
@avasconcelos114 I would go with another alert that shows the error. Maybe @andrewbrown00 can be of help here too.

Reply:
I went ahead and added alerts for both success / error cases. 

The error one uses an `alertErrorWithFallback` function that either displays the response from the server (as the image below), or uses a fallback text that says `We couldn't convert the channel {displayName} to private. Please check your connection and try again.`.  

![image](https://user-images.githubusercontent.com/29700158/64674926-e2708880-d4ac-11e9-9fca-cfa3649dfedc.png)

The alert for success for the time being just says `Succeeded in converting {displayName} to private.`. But I don't think the wording of that is great haha

Let me know what you guys think would be good in there :)